tensorflow_quantum/core/ops/parse_context.h

```diff
@@ -16,6 +16,15 @@ limitations under the License.
 #ifndef TFQ_CORE_OPS_PARSE_CONTEXT
 #define TFQ_CORE_OPS_PARSE_CONTEXT
 
+// Syncs a threads work status with some global status.
+#define NESTED_FN_STATUS_SYNC(global_status, local_status, global_lock) \
+  if (TF_PREDICT_FALSE(!local_status.ok())) {                           \
+    global_lock.lock();                                                 \
+    global_status = local_status;                                       \
+    global_lock.unlock();                                               \
+    return;                                                             \
```

Comment:
Since this is a macro, not a function, should this line be removed?

Reply:
Is it intended to break out of the inner loops?

Reply:
Yes, the macros is intended to break out of the nested function if there was a problem. That's why the return is included.